### ISSUE TITLE
output wizards and sensor types

### ISSUE THREAD

**Issue by tasota** [CMU-CREATE-Lab/flutter-app-android#259]
Currently, when selecting a sensor in the Output Wizards (543db0265492825c8c5427550e72dc252f316ae6) all three sensors are displayed as choices. When a sensor is set as No Sensor/Unknown, it is not selectable. This might present confusion when all 3 sensors are not set since the user will not be able to select anything and cannot click "Next". Any thoughts on giving the user some feedback and prompting them to set their sensor types?

Imagine this screen but with all 3 sensors as "No Sensor" https://app.zeplin.io/project/57f43918d127f7023581eaab/screen/5945c5d8eb69489a44573f73

**Comment by psdev1** [comment]
![device-2018-07-09-155754](https://user-images.githubusercontent.com/13740910/42472913-f76281e4-8390-11e8-83d1-85c3cac6c3a5.png)

The next button is grayed out, and they cannot progress forward in the dialog.

**Comment by jenncross** [comment]
I could imagine a couple of different solutions here: 1) Clicking on an
unset sensor automatically moves you to the sensor tab or pops-up an error
that gives you an option to go to the sensor tab, or 2) instead of
displaying the unset sensors, display a gentle error that says that they
need to set up the sensors first.

The best practice is to avoid dead ends. So instead of getting students
stuck here, give them a nudge in the correct direction to go?

**Comment by tasota** [comment]
I'm trying to brainstorm some ideas this morning as well. I can also see displaying an error-like message when all 3 of these are unset along with a link to the sensors tab.

Alternatively, what if clicking on "No Sensor" at this point brings the pop-up for choosing the sensor type? That way they don't have to be redirected to another tab and click back into Robots tab.

**Comment by jenncross** [comment]
I like both of your solutions:

A) displaying an error-like message when all 3 of these are unset along with a link to the sensors tab
B) clicking on "No Sensor" at this point brings the pop-up for choosing the sensor type

Neither is a dead-end. I have a vague leaning towards A, so that we firmly hold the model that sensors are managed on the sensor tab and emphasize that we want students to explore sensors first.

**Comment by ehamner** [comment]
The new board will auto detect sensors, so if there are no sensors detected then they haven't even plugged one in. I imagine in that case it would have an message to tell them to plug a sensor into the board and then it would detect the sensor and the dialog could continue. What if we plan ahead and make the current version work as close to that as is reasonable?

**Comment by tasota** [comment]
That is true. Does that make the Sensors tab ultimately a bit useless? You won't need to set the sensor types anymore. I think that just leaves pause/resume sensors and simulate sensors for that tab.

**Comment by jenncross** [comment]
The purpose of Sensors Tab is to help kids better visualize the live sensor
values and enable hand recording of values. Encouraging playful
interactions and exploration of data, the setup of sensor there is sort of
a secondary task. So not useless ;-)

The autodetect feature will definitely impact this interaction though. It
might make sense to wait to implement until we see what the capabilities of
the new board are.

I like the idea of reminding someone to plug in a sensor if they try to
select a  relationship type (excluding constant) and no sensors are
detected.
